 Ans 1) To calculate the Pearson correlation coefficient between the amount of time students spend studying for an exam and their final exam scores, you would need to first organize your data into two lists or columns, one for the amount of time spent studying and another for the corresponding final exam score. Once you have your data, you can use a statistical software or calculator that has the capability to compute the Pearson correlation coefficient.

The Pearson correlation coefficient (r) ranges from -1 to +1, where a value of -1 indicates a perfect negative linear relationship between the two variables, 0 indicates no linear relationship, and +1 indicates a perfect positive linear relationship. The closer the value of r is to -1 or +1, the stronger the linear relationship between the two variables.

When interpreting the Pearson correlation coefficient, you should consider the magnitude and sign of the coefficient. A positive coefficient indicates a positive linear relationship, meaning that as the amount of time spent studying increases, so does the final exam score. Conversely, a negative coefficient indicates a negative linear relationship, meaning that as the amount of time spent studying increases, the final exam score decreases. The magnitude of the coefficient indicates the strength of the linear relationship between the two variables, with larger values indicating a stronger relationship.

For example, if the Pearson correlation coefficient between the amount of time spent studying and the final exam score is 0.8, this suggests a strong positive linear relationship between the two variables. This means that students who spend more time studying tend to earn higher scores on the final exam. On the other hand, if the Pearson correlation coefficient is -0.2, this suggests a weak negative linear relationship between the two variables, indicating that students who spend more time studying tend to earn slightly lower scores on the final exam.

Ans 2 ) Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.


To calculate the Spearman's rank correlation between the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10, you would first need to rank each variable separately. Once you have your ranked data, you can use a statistical software or calculator that has the capability to compute the Spearman's rank correlation.

The Spearman's rank correlation coefficient (rho) ranges from -1 to +1, where a value of -1 indicates a perfect negative monotonic relationship between the two variables, 0 indicates no monotonic relationship, and +1 indicates a perfect positive monotonic relationship. Unlike Pearson's correlation coefficient, the Spearman's rank correlation measures the strength and direction of the monotonic relationship, which can be either positive or negative, but not necessarily linear.

When interpreting the Spearman's rank correlation coefficient, you should consider the magnitude and sign of the coefficient. A positive coefficient indicates a positive monotonic relationship, meaning that as the amount of sleep individuals get each night increases, their overall job satisfaction level tends to increase as well. Conversely, a negative coefficient indicates a negative monotonic relationship, meaning that as the amount of sleep individuals get each night increases, their overall job satisfaction level tends to decrease. The magnitude of the coefficient indicates the strength of the monotonic relationship between the two variables, with larger values indicating a stronger relationship.

For example, if the Spearman's rank correlation coefficient between the amount of sleep individuals get each night and their overall job satisfaction level is 0.6, this suggests a strong positive monotonic relationship between the two variables. This means that individuals who get more sleep tend to report higher levels of job satisfaction. On the other hand, if the Spearman's rank correlation coefficient is -0.3, this suggests a weak negative monotonic relationship between the two variables, indicating that individuals who get more sleep tend to report slightly lower levels of job satisfaction.

In [5]:
## Ans 3)
import pandas as pd
import numpy as np
from scipy.stats import pearsonr,spearmanr

#create a sample data with 50 participants
np.random.seed(123)
hours_of_exercise = np.random.normal(5,2,50)
bmi = 25 + 3 * hours_of_exercise + np.random.normal(0,2,50)

# calculate the pearson correlation coefficient
pearson_corr, _ = pearsonr(hours_of_exercise, bmi)
print('Pearson correlation coefficient: %.3f' % pearson_corr)

#calculate the spearman's rank correlation coefficient
df = pd.DataFrame({'hours_of_exercise':hours_of_exercise, 'bmi':bmi})
df['hours_of_exercise_rank'] = df['hours_of_exercise'].rank()
df['bmi_rank'] = df['bmi'].rank()
spearman_corr,_ = spearmanr(df['hours_of_exercise_rank'],df['bmi_rank'])
print('Spearman rank correlation coefficient: %.3f' % spearman_corr)



Pearson correlation coefficient: 0.957
Spearman rank correlation coefficient: 0.948


Ans 3 ) Continue
The Pearson correlation coefficient is calculated as 0.957 indicating a strong  linear relationship between x and y. This means that as the values in x increase, the values in y tend to increase linearly.

The Spearman's rank correlation coefficient is calculated as 0.948, indicating a perfect  monotonic relationship between x and y. This means that as the values in x increase, the values in y tend to increase, but not necessarily in a linear manner.

We can interpret the results as follows:

The Pearson correlation coefficient of 0.957 indicates a strong  linear relationship between x and y.
The Spearman's rank correlation coefficient of 0.948 indicates a perfect  monotonic relationship between x and y.

In [6]:
## Ans 4) 
#Generate random data for hours watching TV and Level of Physical activity
np.random.seed(123)
hours_tv = np.random.normal(3, 1, 50)
activity_level = np.random.normal(5, 2, 50)

#Calculate the Pearson correlation coefficient
pearson_corr,_ = pearsonr(hours_tv, activity_level)

print('Pearson correlation cofficient :%.3f' % pearson_corr)

Pearson correlation cofficient :-0.085


In [9]:
# Ans 5) 
import pandas as pd

df = pd.DataFrame({
    'Soft_drinks' : ['Coke', 'Pepsi', 'Mountain dew', 'Coke', 'Pepsi', 'Coke'],
    'Age' :[25,42,37,19,31,28]
})

In [10]:
df

Soft_drinks  Age
0          Coke   25
1         Pepsi   42
2  Mountain dew   37
3          Coke   19
4         Pepsi   31
5          Coke   28

In [12]:
## Calculate the mean price from softdrinks
mean_price = df.groupby('Soft_drinks')['Age'].mean().to_dict()
mean_price

{'Coke': 24.0, 'Mountain dew': 37.0, 'Pepsi': 36.5}

In [13]:
df['Soft_encoded']=df['Soft_drinks'].map(mean_price)

In [14]:
df

Soft_drinks  Age  Soft_encoded
0          Coke   25          24.0
1         Pepsi   42          36.5
2  Mountain dew   37          37.0
3          Coke   19          24.0
4         Pepsi   31          36.5
5          Coke   28          24.0